### PR TITLE
Also generate cmake usages for pkgconfig files

### PR DESCRIPTION
**Pull request by autoantwort**
Before:
```
➜  vcpkg git:(test-features) ✗ vcpkg x-set-installed --enforce-port-checks 'icu'
Detecting compiler hash for triplet arm64-osx...
All requested packages are currently installed.
Total install time: 1.54 us
```

After:
```
➜  vcpkg git:(test-features) ✗ vcpkg x-set-installed --enforce-port-checks 'icu'
Detecting compiler hash for triplet arm64-osx...
All requested packages are currently installed.
Total install time: 1.58 us
icu can be imported via CMake FindPkgConfig module:

    find_package(PkgConfig)
    # International Components for Unicode: Internationalization library
    pkg_check_modules(icu-i18n REQUIRED IMPORTED_TARGET icu-i18n)
    target_link_libraries(main PkgConfig::icu-i18n)

    # International Components for Unicode: Stream and I/O Library
    pkg_check_modules(icu-io REQUIRED IMPORTED_TARGET icu-io)
    target_link_libraries(main PkgConfig::icu-io)

    # International Components for Unicode: Common and Data libraries
    pkg_check_modules(icu-uc REQUIRED IMPORTED_TARGET icu-uc)
    target_link_libraries(main PkgConfig::icu-uc)

```

### CODE REVIEW

**Comment by dg0yt** [comment]
> ~~~
> pkg_check_modules(icu-uc REQUIRED IMPORTED_TARGET icu-uc)
> target_link_libraries(main PkgConfig::icu-uc)

A good occasion to make a conscious decision: Choose the above, or choose:
> ~~~
> pkg_check_modules(ICU_UC REQUIRED IMPORTED_TARGET icu-uc)
> target_link_libraries(main PkgConfig::ICU_UC)

The first is more legible for the suggested usage. The second makes it clear where the target name comes from, and it is more consistent when also using the variables (`ICU_UC_INCLUDE_DIRS`).

**Comment by BillyONeal** [comment]
Last time the vcpkg maintainers discussed this topic, we opposed adding CMake instructions based on pkgconfig files. For pkgconfig, we should be providing pkgconfig instructions.

In particular, using pkgconfig dependencies like this from CMake can create problems for folks who themselves are trying to ship CMake configs, as the pkgconfig dependency space and the CMake dependency space are disjoint.

See previous discussion https://github.com/microsoft/vcpkg/pull/33001#discussion_r1290694086

@ras0219-msft suggests that the standard should be:

1. If there are CMake configs present, they should be displayed. (Even if there is pkgconfig)
2. If there are pkgconfig files present, they should be displayed. (Even if there are CMake configs)
3. We should not display instructions to try to adapt pkgconfig into CMake.

Also present in discussion: @JavierMatosD @vicroms @data-queue who agreed with the above.

Overall though we agree that helping folks use packages that provide pkgconfig is a good thing, thanks for your contribution there!

**Comment by dg0yt** [comment]
Does this imply that `usage` files should also consider both worlds independently?

**Comment by dg0yt** [comment]
Does this imply that unofficial cmake config is preferred over official pkgconfig?

**Comment by JavierMatosD** [comment]
> Does this imply that usage files should also consider both worlds independently?

Yes

> Does this imply that unofficial cmake config is preferred over official pkgconfig?

Sort of – since our tool is largely CMake-based, the unofficial CMake configs just fit into our workflow more smoothly than the official pkgconfig does. (generated usage should give neither preference over the other)

**Comment by dg0yt** [comment]
I'm just worried about the intrusiveness and lock-in effect of unofficial config, and the detrimental effect on installing and fixing official pkgconfig. 
At least please push upstreaming.

**Comment by BillyONeal** [comment]
> I'm just worried about the intrusiveness and lock-in effect of unofficial config, and the detrimental effect on installing and fixing official pkgconfig.

I think the intent is not 'burn pkgconfig', but 'if we are patching a downstream anyway, and we have the choice between unofficial cmake targets and pkgconfig, we want to choose the unofficial targets, because pkgconfig and cmake configs are separate universes that don't join together cleanly'.

> At least please push upstreaming.

Absolutely!

**Review by BillyONeal — DISMISSED** [review]
(See previous comment)

**Comment by autoantwort** [comment]
Output is now
```
➜  vcpkg git:(test-features) ✗ vcpkg x-set-installed --enforce-port-checks 'icu'
Detecting compiler hash for triplet arm64-osx...
All requested packages are currently installed.
Total install time: 1.67 us
icu provides the following pkgconfig packages:

    # International Components for Unicode: Internationalization library
    icu-i18n

    # International Components for Unicode: Stream and I/O Library
    icu-io

    # International Components for Unicode: Common and Data libraries
    icu-uc

```

File: locales/messages.json [hunk 1]
```diff
@@ -265,6 +265,8 @@
   "CISwitchOptSkipFailures": "Skips ports marked `=fail` in ci.baseline.txt",
   "_CISwitchOptSkipFailures.comment": "=fail is an on-disk format and should not be localized",
   "CISwitchOptXUnitAll": "Reports unchanged ports in the XUnit output",
+  "CMakePkgConfigTargetsUsage": "{package_name} provides the following pkgconfig packages:",
```
Comment (dg0yt) [inline locales/messages.json:268]:
```suggestion
  "CMakePkgConfigTargetsUsage": "{package_name} provides pkg-config modules:",
```

**Review by BillyONeal — COMMENTED** [review]
I added a test for CMake configs; please copy pasta `vcpkg-hello-world-1` or `vcpkg-empty-port`, make it write a .pc you're looking for here, and add to the lists in `usage.ps1`. Then this LGTM I think

**Comment by BillyONeal** [comment]
Couple of nitpicks / questions:

I tried changing your example to `wrong-pkgconfig[core,debug-good,header-only-good,lib-good]`

and now it says:

```console
wrong-pkgconfig provides pkg-config modules:

    # Test lib
    wrong-pkgconfig

    # Test lib
    wrong-pkgconfig
```

which seems wrong?

* Should we say that this result is a heuristic?
* Should the comment derived from 'Description' added after the name of the package rather than before?

**Comment by autoantwort** [comment]
> which seems wrong?

Because the features of the port install pkgconfig files in share/pkgconfig and lib/pkgconfig which no normal port do (you get an postbuildcheck error). 

> Should we say that this result is a heuristic?

But this is not really a heuristic like the cmake one. That are the real pkgconfig modules. 

> Should the comment derived from 'Description' added after the name of the package rather than before?

I could do that. As long as the people don't thing that the `# ` part is part of the module name :D

**Comment by BillyONeal** [comment]
> Because the features of the port install pkgconfig files in share/pkgconfig and lib/pkgconfig which no normal port do (you get an postbuildcheck error).

Makes sense. (I kind of just assumed the feature was named good so it must be good)

> But this is not really a heuristic like the cmake one. That are the real pkgconfig modules.

👍

> I could do that. As long as the people don't thing that the #  part is part of the module name :D

Up to you.

**Comment by autoantwort** [comment]
> Up to you.

I think I prefer the current style :)

**Comment by BillyONeal** [comment]
Thanks for the feature!

**Pull request merged**